src/schema/v2/artwork/index.ts

```diff
@@ -1506,34 +1507,23 @@ const Artwork: GraphQLFieldConfig<void, ResolverContext> = {
       resolveInfo
     )
 
-    const gravityArtworkRes = await artworkLoader(id, gravityParams)
+    const artwork = await artworkLoader(id, gravityParams)
 
-    const enrichedArtwork = {
-      ...gravityArtworkRes,
-    }
-
-    // We don't want to query for the price insights unless the user has requested them
+    // // We don't want to query for the price insights unless the user has requested them
     if (
       marketPriceInsightsBatchLoader &&
-      gravityArtworkRes &&
+      artwork &&
       hasRequestedPriceInsights
     ) {
-      const marketPriceInsight = await marketPriceInsightsBatchLoader([
-        {
-          artistId: gravityArtworkRes.artist._id,
-          medium: gravityArtworkRes.medium,
-          category: gravityArtworkRes.category,
-        },
-      ])
+      const enrichedArtworks = await enrichArtworksWithPriceInsights(
+        [artwork],
+        marketPriceInsightsBatchLoader
+      )
 
-      if (marketPriceInsight.length > 0) {
-        enrichedArtwork.marketPriceInsights = marketPriceInsight[0]
-      } else {
-        enrichedArtwork.marketPriceInsights = null
-      }
+      return enrichedArtworks && enrichedArtworks[0]
```

Comment:
this is probably not necessary, but in case it occurs we will be safe

Reply:
You could simplify this to `enrichedArtworks?.[0]`.

Reply:
😅
true